src/lfp.cpp

```diff
@@ -79,6 +79,44 @@ int lfp_tell(lfp_protocol* f, std::int64_t* n) try {
     return LFP_UNHANDLED_EXCEPTION;
 }
 
+int lfp_peel(lfp_protocol* outer, lfp_protocol** inner) noexcept(true) try{
+    assert(outer);
+    assert(inner);
+
+    auto* tmp = outer->peel();
+    if (!tmp) {
+        *inner = nullptr;
+        outer->errmsg("peel: no underlying protocol");
+        return LFP_IOERROR;
+    }
+
+    *inner = tmp;
+    return LFP_OK;
+} catch(const lfp::error& e) {
+    *inner = nullptr;
+    outer->errmsg(e.what());
+    return e.status();
+}
+
+int lfp_peek(lfp_protocol* outer, lfp_protocol** inner) noexcept(true) try{
+    assert(outer);
+    assert(inner);
+
+    auto* tmp = outer->peek();
+    if (!tmp) {
+        *inner = nullptr;
+        outer->errmsg("peek: no underlying protocol");
+        return LFP_IOERROR;
+    }
+
+    *inner = tmp;
+    return LFP_OK;
+} catch(const lfp::error& e) {
+    outer->errmsg(e.what());
+    *inner = nullptr;
```

Comment:
No - don't overwrite whatever's here unless it's actually good data. Just removing the assignment is fine.